File: flutter/lib/domain/game/horizon.dart
import 'package:ship_conquest/domain/island/island.dart';
import 'package:ship_conquest/domain/space/coordinate.dart';

/// Represents the horizon in game, that is the list of tiles and islands
/// close to the ship.
class Horizon {
  final List<Coordinate> tiles;
  final List<Island> islands;
  // constructor
  Horizon({required this.tiles, required this.islands});
}

File: jvm/ship-conquest/src/main/kotlin/pt/isel/shipconquest/domain/world/Horizon.kt
package pt.isel.shipconquest.domain.world

import pt.isel.shipconquest.domain.space.Vector3
import pt.isel.shipconquest.domain.world.islands.Island


/**
 * The [Horizon] data class holds the data of a fragment from a world, composed
 * of [voxels] and [islands].
 */
data class Horizon(val voxels: List<Vector3>, val islands: List<Island>)